min_heap_binary_tree.rb

```diff
@@ -15,6 +15,27 @@ def insert(node)
     set_min_heap_state
   end
 
+  def delete(node)
+    replacement_node = tree[-1]
+    delete_index = tree.find_index(node)
+    swap! delete_index, -1
```

Comment:
It kind of feels like we could combine lines 21 and 22 into their own private method.  Something like `remove_node`, or `remove_from_tree`, or something like that.